plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hadoop/HadoopIcebergTableOperations.java

```diff
@@ -97,12 +135,257 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
         Map<String, String> properties = table.properties();
         properties.put(METADATA_LOCATION_PROP, metadataLocation);
         properties.put(PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation);
+        commit(base, metadata);
+    }
+
+    public void commit(TableMetadata base, TableMetadata metadata)
+    {
+        Pair<OptionalInt, TableMetadata> current = Pair.of(version, currentMetadata);
+        if (base != current.second()) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, "Cannot commit changes based on stale table metadata");
+        }
+
+        checkArgument(
+                base == null || base.location().equals(metadata.location()),
+                "Hadoop path-based tables cannot be relocated");
+        checkArgument(
+                !metadata.properties().containsKey(TableProperties.WRITE_METADATA_LOCATION),
+                "Hadoop path-based tables cannot relocate metadata");
```

Comment:
Are these two errors supposed to mean the same thing? Asking for future self just in case

Reply:
there are not quite exactly the same , one message is for the data table itself and one is for the table containing its metadata so it means you can`t relocate a table and you can't relocate its metadata table (which the data table information)